core/store/src/cold_storage.rs

```diff
@@ -136,6 +144,38 @@ fn get_keys_from_store(
             key_type,
             match key_type {
                 DBKeyType::BlockHash => vec![block_hash_key.clone()],
+                DBKeyType::ShardUId => shard_layout
+                    .get_shard_uids()
+                    .iter()
+                    .map(|uid| uid.to_bytes().to_vec())
+                    .collect(),
+                DBKeyType::TrieNodeOrValueHash => {
+                    let mut keys = vec![];
+                    for shard_uid in shard_layout.get_shard_uids() {
+                        let shard_uid_key = shard_uid.to_bytes().to_vec();
```

Comment:
If you change functions to take `&[u8]` rather than `&StoreKey`, you won’t need to allocate vector here.
```suggestion
                        let shard_uid_key = shard_uid.to_bytes();
```